mindsdb/integrations/handlers/writer_handler/evaluator.py

```diff
@@ -0,0 +1,13 @@
+from scipy.spatial import distance
+
+
+def calculate_cosine_similarity(
+    context_embeddings: list, retrieved_embeddings: list
+) -> float:
+    cosine_sim = 1 - distance.cosine(context_embeddings, retrieved_embeddings)
+
+    return cosine_sim
+
+
+def accuracy(cosine_similarity: float, threshold: float = 0.7) -> bool:
+    return cosine_similarity >= threshold
```

Comment:
I propose we use a threshold to indicate whether a successful match has happened, we need to set this fairly low to ensure it accounts for matches with are a subset of the full context